HungryWolfs/Repository/Repository.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  Repository.swift
+//  HungryWolfs
+//
+//  Created by Wolfpack Digital on 11.08.2021.
+//
+
+import Foundation
+import Alamofire
+
+class Repository
+{
+    
+    let myFoodURL = "https://www.themealdb.com/api/json/v1/1/categories.php"
+    
+    func getCategories(completion: @escaping ([Food]?) -> Void)
+    {
+        AF.request(myFoodURL).responseJSON
+        { response in
+            switch response.result
+            {
+            case .success:
+                let decoder = JSONDecoder()
+                guard let data = response.data else { return }
+                let categoryArray = try? decoder.decode(FoodCategories.self, from: data)
+                completion(categoryArray?.categories)
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+    
+    
+    func getMeals(categories: String, completion: @escaping ([Meal]?) -> Void)
+    {
+        let myMealURL = "https://www.themealdb.com/api/json/v1/1/filter.php?c=" + categories
+        
+        AF.request(myMealURL).responseJSON
+        { response in
+            switch response.result
+            {
+            case .success:
+                let decoder = JSONDecoder()
+                guard let data = response.data else { return }
+                let mealsArray = try? decoder.decode(FoodMeals.self, from: data)
+                completion(mealsArray?.meals)
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+    
+    func getSearch(name: String, completion: @escaping ([Meal]?) -> Void)
+    {
+        let myMealURL = "https://www.themealdb.com/api/json/v1/1/search.php?s=" + name
```

Comment:
You can extract the part of the URL that's common to all functions (https://www.themealdb.com/api/json/v1/1/) to a constant, then use it in each function by only appending the suffix to it (in this case "search.php?s= ... etc)